src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -458,13 +309,6 @@ private void FreeNativeResources()
             }
         }
 
-        private bool TryRegister(SafeSocketHandle socket, IntPtr handle, out Interop.Error error)
```

Comment:
Could you create a GCHandle for the `SocketAsyncContextWrapper`, register the GCHandle as data against the socket and then use (SocketAsyncContextWrapper)(Handle.Target) rather than the `ConcurrentDictionary<IntPtr, SocketAsyncContextWrapper>`?